src/pages/workspace/accounting/netsuite/import/NetSuiteImportCustomFieldNew/NetSuiteImportAddCustomSegmentPage.tsx

```diff
@@ -0,0 +1,220 @@
+import withPolicyConnections from '@pages/workspace/withPolicyConnections';
+import CONST from '@src/CONST';
+import type {TranslationPaths} from '@src/languages/types';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import INPUT_IDS from '@src/types/form/NetSuiteCustomFieldForm';
+import ChooseSegmentTypeStep from './substeps/ChooseSegmentTypeStep';
+import ConfirmCustomSegmentList from './substeps/ConfirmCustomSegmentList';
+import CustomSegmentInternalIdStep from './substeps/CustomSegmentInternalIdStep';
+import CustomSegmentNameStep from './substeps/CustomSegmentNameStep';
+import CustomSegmentScriptIdStep from './substeps/CustomSegmentScriptIdStep';
+import MappingStep from './substeps/MappingStep';
+
+const formSteps = [ChooseSegmentTypeStep, CustomSegmentNameStep, CustomSegmentInternalIdStep, CustomSegmentScriptIdStep, MappingStep, ConfirmCustomSegmentList];
+
+function NetSuiteImportAddCustomSegmentPage({policy}: WithPolicyConnectionsProps) {
+    const policyID = policy?.id ?? '-1';
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const ref: ForwardedRef<InteractiveStepSubHeaderHandle> = useRef(null);
+
+    const config = policy?.connections?.netsuite?.options?.config;
+    const customSegments = useMemo(() => config?.syncOptions?.customSegments ?? [], [config?.syncOptions]);
+
+    const handleFinishStep = useCallback(() => {
+        Navigation.goBack(ROUTES.POLICY_ACCOUNTING_NETSUITE_IMPORT_CUSTOM_FIELD_MAPPING.getRoute(policyID, CONST.NETSUITE_CONFIG.IMPORT_CUSTOM_FIELDS.CUSTOM_SEGMENTS));
+    }, [policyID]);
+
+    const {
+        componentToRender: SubStep,
+        isEditing,
+        nextScreen,
+        prevScreen,
+        screenIndex,
+        moveTo,
+        goToTheLastStep,
+    } = useSubStep<CustomFieldSubStepWithPolicy>({bodyContent: formSteps, startFrom: CONST.NETSUITE_CUSTOM_FIELD_SUBSTEP_INDEXES.CUSTOM_SEGMENTS.SEGMENT_TYPE, onFinished: handleFinishStep});
+
+    const handleBackButtonPress = () => {
+        if (isEditing) {
+            goToTheLastStep();
+            return;
+        }
+
+        // Clicking back on the first screen should go back to listing
+        if (screenIndex === CONST.NETSUITE_CUSTOM_FIELD_SUBSTEP_INDEXES.CUSTOM_SEGMENTS.SEGMENT_TYPE) {
+            Navigation.goBack(ROUTES.POLICY_ACCOUNTING_NETSUITE_IMPORT_CUSTOM_FIELD_MAPPING.getRoute(policyID, CONST.NETSUITE_CONFIG.IMPORT_CUSTOM_FIELDS.CUSTOM_SEGMENTS));
+            return;
+        }
+        ref.current?.movePrevious();
+        prevScreen();
+    };
+
+    const handleNextScreen = useCallback(() => {
+        if (isEditing) {
+            goToTheLastStep();
+            return;
+        }
+        ref.current?.moveNext();
+        nextScreen();
+    }, [goToTheLastStep, isEditing, nextScreen]);
+
+    const [customSegmentType, setCustomSegmentType] = useState<ValueOf<typeof CONST.NETSUITE_CUSTOM_RECORD_TYPES> | undefined>();
+
+    const validate = useCallback(
```

Comment:
Some validation was missing here which caused issue:
- https://github.com/Expensify/App/issues/46536

To solve the issue we added new error case and translations for the specific case, more details in the [proposal](https://github.com/Expensify/App/issues/46536#issuecomment-2269373151).